### PR TITLE
Feat: Supported gcm device registration

### PR DESCRIPTION
**Pull request by FrankenSteinxD**
add a "platform" query param to the request to decide the platform, valid options are [gcm|apns]

### CODE REVIEW

File: registration/registration.go [hunk 1]
```diff
@@ -22,21 +22,40 @@ type registrationPayload struct {
 	ID    string
 }
 
+// APNS
```
Comment (mhd999) [inline registration/registration.go]: no need for this comment

File: registration/registration.go [hunk 2]
```diff
@@ -22,21 +22,40 @@ type registrationPayload struct {
...
-	XMLName xml.Name `xml:"entry"`
-	Xmlns   string   `xml:"xmlns,attr"`
-	Content Content  `xml:"content"`
+type EntryAPNS struct {
+	XMLName xml.Name 	`xml:"entry"`
+	Xmlns   string   	`xml:"xmlns,attr"`
+	Content ContentAPNS `xml:"content"`
+}
+
+// GCM
```
Comment (mhd999) [inline registration/registration.go]: no need for this comment

File: registration/registration.go [hunk 3]
```diff
@@ -56,6 +75,36 @@ func createRegistrationID(sasToken string, hubNamespace string, hubName string)
 	return registrationID
 }
 
+func GetXmlBody(platform string, payload registrationPayload) ([]byte, error) {
```
Comment (mhd999) [inline registration/registration.go]: `([]byte, error)` can't see what the case this function will return error. i think it should only return `([]byte)`

File: registration/registration.go [hunk 4]
```diff
@@ -95,19 +144,14 @@ func CreateOrUpdateRegistration(w http.ResponseWriter, r *http.Request) {
...
-		Xmlns: "http://www.w3.org/2005/Atom",
-		Content: Content{
-			Type: "application/xml",
-			AppleRegistrationDescription: AppleRegistrationDescription{
-				Xmlns:       "http://schemas.microsoft.com/netservices/2010/10/servicebus/connect",
-				DeviceToken: registrationPayload.Token,
-				Tags:        strings.Join(registrationPayload.Tags[:], ","),
-			},
-		},
+	query := r.URL.Query()
```
Comment (mhd999) [inline registration/registration.go]: can you move this logic up in the function, to line 129 so we can avoid generate a token if the platform query doesn't exist

File: registration/registration.go [hunk 5]
```diff
@@ -95,19 +144,14 @@ func CreateOrUpdateRegistration(w http.ResponseWriter, r *http.Request) {
...
+	platform, ok := query["platform"]
+	if !ok {
+		http.Error(w, err.Error(), 500)
+		return
```
Comment (mhd999) [inline registration/registration.go]: can you move this logic up in the function, to line 129 so we can avoid generate a token if the platform query doesn't exist

File: registration/registration.go (hunk 3 above)
Comment (FrankenSteinxD) [reply registration/registration.go]: xml.MarshaIndet returns ([]byte, error), and I am returning xml.MarshalIndet ==> I have to write it :\

**Pull request merged**